audit.js

```diff
@@ -0,0 +1,56 @@
+const shimmer = require('shimmer')
+const assert = require('assert')
+const os = require('os')
+const uuidv4 = require('uuid/v4')
+
+module.exports = function (context, opts) {
+  assert(opts, 'Options object must be defined')
+  assert(typeof opts.eventName === 'string', 'Event name must be a string')
+  assert(typeof opts.resource === 'string', 'Resource must be a string')
+
+  context['audit-options'] = opts
+  context['audit-queue'] = []
+  context['audit-system'] = `${os.hostname()}-${os.platform()}`
+
+  context['audit'] = extensionLogArgs => {
+    const { eventName, resource } = context['audit-options']
+
+    context['audit-queue'].push({
```

Comment:
if we're going to keep this state on context, it should at least be escaped/unlikely to collide with user/platform features - like we did with `body['__audits']`